src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -46,4 +55,24 @@
   }
 
 
+  /**
+     * Get the historical price for a symbol.
+     * Define required parameters. word
+     */
+
+  @Cacheable(cacheNames = "HistoricalPrice", sync = true)
+  public List<IexHistoricalPrice> getHistoricalPriceForSymbols(
+      final String symbol,
+      final String range,
+      final String date,
+      final String token) {
+    if (CollectionUtils.isEmpty(Collections.singleton(symbol))) {
```

Comment:
Also this check is invalid. The collection will only be empty if symbol is null.
Maybe take a look at ```StringUtils```